src/components/Sidebar.tsx

```diff
@@ -231,9 +289,41 @@ const SubSiderBar = ({
   return null;
 };
 
-const Sidebar = ({ sidebarNodes }: ISidebarProps) => {
+const Sidebar = ({
+  engine,
+  sidebarNodes,
+  updateSelectedBar,
+}: ISidebarProps) => {
   const [selected, setSelected] = React.useState<string | null>(null);
 
+  const state = engine
+    .getStateMachine()
+    .getCurrentState() as DefaultDiagramState;
+
+  const enableDrag = () => {
+    if (!state.dragCanvas.config.allowDrag) {
+      state.dragCanvas.config.allowDrag = true;
+      updateCanvasMouseCursor('move');
+    }
+  };
+
+  const disableDrag = () => {
+    if (state.dragCanvas.config.allowDrag) {
+      state.dragCanvas.config.allowDrag = false;
+      updateCanvasMouseCursor('default');
+    }
+  };
+
+  const updateSelected = (id: string) => {
+    setSelected(id);
+    if (updateSelectedBar) updateSelectedBar(id);
+  };
+
+  useEffect(() => {
+    disableDrag();
+    updateCanvasMouseCursor('default');
```

Comment:
Assuming state.dragCanvas.config.allowDrag is true, which seems like it is from the [code](https://github.com/projectstorm/react-diagrams/blob/master/packages/react-canvas-core/src/states/DragCanvasState.ts#L22), we may not need this instruction

Reply:
I would recommend to replace ['magic strings'](https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings) like 'default' for a variable.
By doing that you make it easier to change it later on without the need to go into all the multiple places it could be being used on